src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignPriceEstimateContext.tsx

```diff
@@ -100,41 +147,28 @@ function getActions(dispatch, relayEnvironment) {
             $medium: String!
           ) {
             marketPriceInsights(artistId: $artistInternalID, medium: $medium) {
-              annualLotsSold
-              annualValueSoldCents
-              artistId
               artistName
-              artsyQInventory
-              createdAt
-              demandRank
-              demandTrend
               highRangeCents
-              largeHighRangeCents
-              largeLowRangeCents
-              largeMidRangeCents
-              liquidityRank
               lowRangeCents
-              medianSaleToEstimateRatio
-              medium
-              mediumHighRangeCents
-              mediumLowRangeCents
-              mediumMidRangeCents
-              midRangeCents
-              sellThroughRate
-              smallHighRangeCents
-              smallLowRangeCents
-              smallMidRangeCents
-              updatedAt
             }
           }
         `,
         {
-          artistInternalID,
+          artistInternalID: selectedSuggestion.node.internalID,
           // FIXME: Figure out how to handle this constraint
           medium: "PAINTING",
         }
       )
 
+      actions.setFetching(false)
+
+      dispatch({
+        type: "selectedSuggestion",
+        payload: {
+          selectedSuggestion,
+        },
+      })
```

Comment:
There's a reason why this was moved into this function: the image in the main area comes from the search result query (elastic search), but the estimate data comes from diffusion. When a user performs a search and clicks a result, the query is already complete, so the image updates immediately while the diffusion data is being fetched. This is weird UX, so by deferring the selected suggestion state update till _after_ the diffusion fetch is complete we update all state at once leading to a react lifecyle refresh that's unified rather than staggered, which makes for more stable UI.

Reply:
Why wouldn't an `await` in `actions.selectSuggestion` fix this problem? E.g.:
```
selectSuggestion: selectedSuggestion => {
  await actions.fetchArtistInsights(selectedSuggestion)
  dispatch({
        type: "selectedSuggestion",
        payload: {
          selectedSuggestion,
        },
      })
}
```

Reply:
Ahh great suggestion, that solves this issue nicely. Will fix